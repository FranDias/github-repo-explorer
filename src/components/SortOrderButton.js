import React, { Component } from "react";
import PropTypes from "prop-types";

class SortOrderButton extends Component {
  static propTypes = {
    sortDirection: PropTypes.oneOf([undefined, "asc", "dec"]),
    onClick: PropTypes.func.isRequired
  };

  static defaultProps = {
    sortDirection: undefined
  }

  renderArrowDirection() {
    const { sortDirection } = this.props;
    if (!sortDirection) return "•"

    return this.props.sortDirection === "asc" ? "⬆" : "⬇"
  }

  render() {
    return <button onClick={this.props.onClick}>{this.renderArrowDirection()}</button>;
  }
}
        
export default SortOrderButton;