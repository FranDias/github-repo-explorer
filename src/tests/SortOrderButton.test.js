import React from "react";
import renderer from "react-test-renderer";
import SortOrderButton from "../components/SortOrderButton";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("SortOrderButton", () => {
  let component;
  const testFunction = jest.fn();

  it("matches its snapshot with minimal props", () => {
    const component = renderer.create(<SortOrderButton onClick={() => {}} />);
    expect(component).toMatchSnapshot();
  });

  beforeEach(() => {
    component = shallow(<SortOrderButton onClick={() => {}} />);
  });

  it("renders a • when no sort order is given", () => {
    expect(component.find("button").text()).toEqual("•");
  });

  it('renders a "⬆" with sort order asc', () => {
    component.setProps({ sortDirection: "asc" });
    expect(component.find("button").text()).toEqual("⬆");
  });

  it('renders a "⬇" with sort order dec', () => {
    component.setProps({ sortDirection: "dec" });
    expect(component.find("button").text()).toEqual("⬇");
  });

  it("calls onClick when clicked", () => {
    const testFunction = jest.fn();
    component.setProps({ onClick: testFunction });
    component.simulate("click");
    expect(testFunction).toHaveBeenCalled();
  });
});
