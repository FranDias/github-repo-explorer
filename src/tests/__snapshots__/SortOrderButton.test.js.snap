// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortOrderButton matches its snapshot with minimal props 1`] = `
<button
  onClick={[Function]}
>
  â€¢
</button>
`;
